docs/topics/Gateway.md

```diff
@@ -120,12 +120,11 @@ Once connected, the client should immediately receive an [Opcode 10 Hello](#DOCS
 
 ### Heartbeating
 
-The client should now begin sending [Opcode 1 Heartbeat](#DOCS_TOPICS_GATEWAY/heartbeat) payloads every `heartbeat_interval` milliseconds, until the connection is eventually closed or terminated. This OP code is also bidirectional. The gateway may request a heartbeat from you in some situations, and you should send a heartbeat back to the gateway as you normally would.
+After receiving [Opcode 10 Hello](#DOCS_TOPICS_GATEWAY/hello), the client may begin sending [Opcode 1 Heartbeat](#DOCS_TOPICS_GATEWAY/heartbeat) payloads at any time before the first `heartbeat_inverval` period elapses. After this, `heartbeat_interval` dictates the *maximum* time between any two heartbeats. You may send heartbeats before this interval elapses, but you should avoid doing so unless necessary. There is already tolerance in the `heartbeat_interval` that will cover network latency, so you do not need to account for it in your own implementation - waiting the precise interval will suffice.
```

Comment:
this explanation gives too much ambiguity and flexibility. the reason the original phrasing worked is because it told a user exactly what to do for heartbeating.

while there's some original ambiguity around when to begin the heartbeat, we should instead just clarify that it should be sent `after heartbeat_interval millseconds, and every heartbeat_interval milliseconds thereafter`

Reply:
Sure, that sounds fine to me. The reason I proposed this, is @jhgg (iirc) once gave an explanation that the desktop client waits a random amount of time less than the interval before starting to heartbeat, which helps prevent bursts of traffic at `heartbeat_interval` periods when reconnecting waves of sessions at once on Discord's end.

I recognize that bots are only a small subset of this traffic, and the same thinking likely doesn't matter as much - so I'm happy to reword this if we want to stick with "wait until the first interval elapses". It's still an improvement.

I can reword this later tonight if there is no further feedback by then.

Reply:
if we want to support that use case for bots too, we can provide sample code involving `random` like

```
after `heartbeat_interval * random.random()` milliseconds
```

Reply:
Why not just say something like

> After receiving the `heartbeat_interval`, you can start sending heartbeats. You can send the first heartbeat **within** `heartbeat_interval`, or wait until the interval has passed to start.

I feel like documenting it to this level of granularity risks people introducing bugs if Discord does decide to change this slightly in the future.